flink-runtime/src/main/java/org/apache/flink/runtime/state/changelog/ChangelogStateBackendHandle.java

```diff
@@ -48,20 +49,25 @@
 
     List<ChangelogStateHandle> getNonMaterializedStateHandles();
 
+    UUID getBackendID();
```

Comment:
What the purpose of UUID acts here?

Reply:
UUID identifies the backend, which allows to avoid collisions of state id across different backends on recovery (e.g. in case of downscaling).

Reply:
Thanks for your explanation. I think we could add a basic java doc on it to describe this.